src/arch/riscv/tlb.cc

```diff
@@ -366,6 +383,15 @@ TLB::translate(const RequestPtr &req, ThreadContext *tc,
 
         if (!delayed && fault == NoFault) {
             pma->check(req);
+            // do pmp checks here if not in M mode
+            if (pmode != PrivilegeMode::PRV_M){
```

Comment:
Could you make the `pmpCheck` function return a fault (or `NoFault`)?

I see that the code below is duplicated here and in the walker. Can we move it to the `PMP` object?

Reply:
done. These functions are moved to the PMP class